maintainer/update_json_stubs_sitemap.py

```diff
@@ -117,10 +117,55 @@ def write_redirect(file, version='', outfile=None):
 # latest/index.html -> latest release (not dev docs)
 # stable/ : a copy of the release docs with the highest number (2.0.0 instead of 1.0.0)
 # dev/ : a copy of the develop docs with the highest number (2.0.0-dev instead of 1.0.1-dev)
+# sitemap.xml files are updated by replacing URL strings
 
+
+def redirect_sitemap(old_version, new_version):
+    """Replace paths in copied sitemap.xml with new directory path
+
+    Sitemaps can only contain URLs 'within' that directory structure.
+    For more, see https://www.sitemaps.org/faq.html#faq_sitemap_location
+    """
+    file = f"{new_version}/sitemap.xml"
+    old = f"{URL}/{old_version}/"
+    new = f"{URL}/{new_version}/"
+    try:
+        with open(file, "r") as f:
+            contents = f.read()
+    except OSError:
+        raise ValueError(f"{file} not found")
+    redirected = contents.replace(old, new)
```

Comment:
Did you consider working with the XML instead?

Reply:
In the sense of replacing the URLs for each element separately? As that's still string manipulation I thought just replacing everything in the text file would be easier.